core/client/fs/src/main/java/alluxio/client/file/FileSystemUtils.java

```diff
@@ -143,14 +145,16 @@ public static void persistAndWait(final FileSystem fs, final AlluxioURI uri)
    *
    * @param fs {@link FileSystem} to carry out Alluxio operations
    * @param uri the uri of the file to persist
+   * @param persistenceWaitTime the initial persistence wait time
    * @param timeoutMs max amount of time to wait for persist in milliseconds. -1 to wait
    *                  indefinitely
    * @throws TimeoutException if the persist takes longer than the timeout
    */
-  public static void persistAndWait(final FileSystem fs, final AlluxioURI uri, int timeoutMs)
-      throws FileDoesNotExistException, IOException, AlluxioException, TimeoutException,
-      InterruptedException {
-    fs.persist(uri);
+  public static void persistAndWait(final FileSystem fs, final AlluxioURI uri,
+      long persistenceWaitTime, int timeoutMs) throws FileDoesNotExistException, IOException,
```

Comment:
To help prevent user error, can we check that `persistenceWaitTime < timeoutMs`? When that condition is true, the persist will always timeout and fail here. It would be useful to fail-fast by checking this condition.

Reply:
Add the preconditions check here. Thanks